key_gen.go

```diff
@@ -57,17 +79,14 @@ func KeyForPrefixedStringMap(
 	return keyForPrefixedStringMaps(prefix, stringMap)
 }
 
-// keyForPrefixedStringMaps generates a unique key for a prefix and a series
-// of maps containing tags.
-//
-// If a key occurs in multiple maps, keys on the right take precedence.
-func keyForPrefixedStringMaps(prefix string, maps ...map[string]string) string {
-	keys := keyGenPool.stringsPool.Get().([]string)
+func keyForPrefixedStringMapsAsKey(prefix string, maps ...map[string]string) key {
+	keys := keyGenPool.stringsPool.Get()
 	for _, m := range maps {
 		for k := range m {
 			keys = append(keys, k)
 		}
 	}
+	// 1 allocation as sort.Interface escapes to heap
```

Comment:
Interesting, somewhat surprising, and a little bit disappointing. :-)  It will likely be helped by the generics in Go 1.18, where you don't need the boxing as much (hopefully).